src/sage/knots/knotinfo.py

```diff
@@ -1987,36 +1996,47 @@ def khovanov_polynomial(self, var1='q', var2='t', base_ring=ZZ, original=False,
             1
             sage: K0_1.khovanov_polynomial(odd=True)
             1
-            sage: K0_1.khovanov_polynomial(base_ring=GF(3), reduced=True)
+            sage: K0_1.khovanov_polynomial(ring=GF(3), reduced=True)
             Traceback (most recent call last):
             ...
-            ValueError: characteristic 3 of base ring is not valid
-            sage: K0_1.khovanov_polynomial(base_ring=GF(3), odd=True)
+            ValueError: characteristic 3 of ring is not valid
+            sage: K0_1.khovanov_polynomial(ring=GF(3), odd=True)
             Traceback (most recent call last):
             ...
-            ValueError: characteristic 3 of base ring is not valid
-            sage: L.khovanov_polynomial(base_ring=GF(2))
+            ValueError: characteristic 3 of ring is not valid
+            sage: L.khovanov_polynomial(ring=GF(2))
             Traceback (most recent call last):
             ...
-            NotImplementedError: Khovanov polynomial available only for knots in characteristic 2
+            NotImplementedError: Khovanov polynomial not available for multi-component links for this ring
 
         REFERENCES:
 
         - :wikipedia:`Khovanov_homology`
         - :wikipedia:`Reduced_homology`
         - [ORS2013]_
+        - `KnotInfo <https://knotinfo.math.indiana.edu/descriptions/khovanov_unreduced_integral_polynomial.html>`__
         """
-        ch = base_ring.characteristic()
-        integral = ch == 0 and base_ring.is_field()
+        if not ring:
+            if self.is_knot():
+                ring = ZZ
+            else:
+                from sage.rings.rational_field import QQ
+                ring = QQ
+
+        if base_ring:
+            ring = base_ring
+            from sage.misc.superseded import deprecation
+            deprecation(40149, "base_ring is deprecated, use argument ring instead.")
+
+        ch = ring.characteristic()
+        integral = ch == 0 and not ring.is_field()
```

Comment:
```suggestion
        integral = (ch == 0) and (not ring.is_field())
```
While not necessary, I think it makes it easier to parse.